File: kys_formats/role_meta.py
"""Role field labels and enums aligned with Role.h / TRole."""

from __future__ import annotations

import struct


def as_u16(signed_word: int) -> int:
    return int(signed_word) & 0xFFFF


def to_i16_from_u16(value: int) -> int:
    return struct.unpack("<h", struct.pack("<H", int(value) & 0xFFFF))[0]


SEXUAL = {
    0: "男",
    1: "女",
}

MP_TYPES = {
    0: "阴",
    1: "阳",
    2: "调和",
}

EQUIP_SLOTS = {
    0: "武器",
    1: "身披",
    2: "头戴",
    3: "脚踩",
    4: "第五装备位",
}


def sexual_display(v: int) -> str:
    return SEXUAL.get(v, f"未知({v})")


def mp_type_display(v: int) -> str:
    return MP_TYPES.get(v, f"未知({v})")


def role_summary(rec: list, name: str = "") -> str:
    """Short summary for list rows."""
    lv = rec[15] if len(rec) > 15 else 0
    sex = sexual_display(rec[14] if len(rec) > 14 else 0)
    return f"{name}  Lv{lv} {sex}"

File: for_Android/app/src/main/java/com/kys/editor/codec/meta/RoleMeta.kt
package com.kys.editor.codec.meta

object RoleMeta {
    val SEXUAL: Map<Int, String> = mapOf(
        0 to "男",
        1 to "女"
    )

    val MP_TYPES: Map<Int, String> = mapOf(
        0 to "阴",
        1 to "阳",
        2 to "调和"
    )

    val EQUIP_SLOTS: Map<Int, String> = mapOf(
        0 to "武器",
        1 to "身披",
        2 to "头戴",
        3 to "脚踩",
        4 to "第五装备位"
    )

    fun asU16(signedWord: Int): Int = signedWord and 0xFFFF

    fun toI16FromU16(value: Int): Short {
        val v = value and 0xFFFF
        return if (v >= 0x8000) (v - 0x10000).toShort() else v.toShort()
    }

    fun sexualDisplay(v: Int): String = SEXUAL[v] ?: "未知($v)"
    fun mpTypeDisplay(v: Int): String = MP_TYPES[v] ?: "未知($v)"
    fun equipSlotDisplay(v: Int): String = EQUIP_SLOTS[v] ?: "未知($v)"

    fun roleSummary(rec: IntArray, name: String = ""): String {
        val lv = if (rec.size > 15) rec[15] else 0
        val sex = sexualDisplay(if (rec.size > 14) rec[14] else 0)
        return "$name  Lv$lv $sex"
    }
}
